src/Workspaces/Core/Portable/FlowAnalysis/ReachingDefinitions/DefinitionUsageResult.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.FlowAnalysis.ReachingDefinitions
+{
+    internal sealed class DefinitionUsageResult
+    {
+        public DefinitionUsageResult(
+            ImmutableDictionary<(ISymbol Symbol, IOperation Definition), bool> definitionUsageMap,
+            ImmutableHashSet<ISymbol> symbolsRead)
+        {
+            DefinitionUsageMap = definitionUsageMap;
+            SymbolsRead = symbolsRead;
+        }
+
+        /// <summary>
+        /// Map from each symbol definition to a boolean indicating if the value assinged
+        /// at definition is used/read on some control flow path.
+        /// </summary>
+        public ImmutableDictionary<(ISymbol Symbol, IOperation Definition), bool> DefinitionUsageMap { get; }
```

Comment:
1. i'm a bit confused how this is keyed by symbol/definition.  What does that mean?
2. tuple names are lowercase in IDE.